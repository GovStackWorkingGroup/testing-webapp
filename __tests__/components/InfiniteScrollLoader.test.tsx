import InfiniteScrollCustomLoader from '../../components/InfiniteScrollLoader';
import { render } from '../test-utils/test-utils';

describe('Unit tests for InfiniteScrollCustomLoader component:', () => {
  const SCROLL_LOADER_TEST_ID = 'scroll-loader';

  it('should render the InfiniteScrollCustomLoader component and match snapshot', () => {
    const { container } = render(<InfiniteScrollCustomLoader />);

    expect(container).toMatchSnapshot();
  });

  it('should have proper value', () => {
    const { getByTestId } = render(<InfiniteScrollCustomLoader />);

    expect(getByTestId(SCROLL_LOADER_TEST_ID)).toHaveTextContent('Loading more data...');
  });
});
