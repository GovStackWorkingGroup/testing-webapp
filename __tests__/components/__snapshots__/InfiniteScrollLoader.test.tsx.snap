// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for InfiniteScrollCustomLoader component: should render the InfiniteScrollCustomLoader component and match snapshot 1`] = `
<div>
  <div
    class="data-loader-message"
    data-testid="scroll-loader"
  >
    <div
      class="spinner"
    >
      <div
        class="loader"
      />
    </div>
     
    Loading more data...
  </div>
</div>
`;
